packages/loopring_v3/circuit/Gadgets/MathGadgets.h

```diff
@@ -1806,7 +1807,7 @@ class SelectGadget : public GadgetT
         for (unsigned int i = 0; i < values.size(); i++)
         {
             results.emplace_back(TernaryGadget(
-              pb, selector[i], values[i], (i == 0) ? values[0] : results.back().result(), FMT(prefix, ".results")));
+              pb, selector[i], values[i], (i == 0) ? _constants._0 : results.back().result(), FMT(prefix, ".results")));
```

Comment:
Just out of curiosity, does this save constraints?

Reply:
No, it's a workaround to make our trusted setup work. The specific value is unimportant here, and using `values[0]` makes the trusted setup proving keys not work for reasons that are not completely clear to me. I think there's an issue for some reason when using the same variable multiple times in a linear combination in a constraint, but totally not sure about that.